Cannot use `PYAPP_DISTRIBUTION_PATH` (at least with locally embedded projects)

When embedding a local tar of a python distribution, I get the following error

```
  --- stderr
  thread 'main' panicked at build.rs:565:9:

  Unable to determine format for distribution source: 
```

(yes, `distribution_source` is an empty string, I checked).

This occurs when providing `PYAPP_PROJECT_PATH` (embedding a wheel), `PYAPP_DISTRIBUTION_PATH`, and `PYAPP_EXEC_SPEC`. Removing `PYAPP_DISTRIBUTION_PATH`, works but python is not bundled along with the binary.

I have an [example here](https://github.com/nmay231/pyapp_bug).

<details>

<summary>

full output of `make build-bugged`

</summary>

```
$ make build-bugged
poetry build -f wheel
Building pyapp-bug (0.1.0)
  - Building wheel
  - Built pyapp_bug-0.1.0-py3-none-any.whl
cd pyapp-latest && \
	PYAPP_PROJECT_PATH=../dist/pyapp_bug-0.1.0-py3-none-any.whl \
	PYAPP_DISTRIBUTION_PATH=./python-3.10.13.tar.gz \
	PYAPP_EXEC_SPEC=pyapp_bug.entry:main \
	cargo build --release
   Compiling pyapp v0.13.0 (/home/nmay/cur/pyapp_bug/pyapp-latest)
error: failed to run custom build command for `pyapp v0.13.0 (/home/nmay/cur/pyapp_bug/pyapp-latest)`

Caused by:
  process didn't exit successfully: `/home/nmay/cur/pyapp_bug/pyapp-latest/target/release/build/pyapp-fa2006654aeb24f9/build-script-build` (exit status: 101)
  --- stdout
  cargo:rustc-env=PYAPP_PROJECT_DEPENDENCY_FILE=
  cargo:rustc-env=PYAPP__PROJECT_DEPENDENCY_FILE_NAME=
  cargo:rustc-env=PYAPP_PROJECT_NAME=pyapp-bug
  cargo:rustc-env=PYAPP_PROJECT_VERSION=0.1.0
  cargo:rustc-env=PYAPP__PROJECT_EMBED_FILE_NAME=pyapp_bug-0.1.0-py3-none-any.whl
  cargo:rustc-env=PYAPP_DISTRIBUTION_SOURCE=
  cargo:rustc-env=PYAPP__DISTRIBUTION_ID=12080488350626410958

  --- stderr
  thread 'main' panicked at build.rs:565:9:


  Unable to determine format for distribution source: 


  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
make: *** [Makefile:13: build-bugged] Error 101
```

</details>

--- a/.gitignore
+++ b/.gitignore
@@ -20,3 +20,6 @@ Cargo.lock
 # Documentation
 /.cache/
 /site/
+
+# Editor configs
+.vscode/

--- a/build.rs
+++ b/build.rs
@@ -502,7 +502,7 @@ fn set_distribution() {
         let mut file = File::open(&embed_path).unwrap();
         std::io::copy(&mut file, &mut hasher).unwrap();
 
-        "".to_string()
+        local_path
     } else if is_enabled("PYAPP_DISTRIBUTION_EMBED") {
         let distribution_source = get_distribution_source();
         let bytes = reqwest::blocking::get(&distribution_source)
@@ -678,6 +678,9 @@ fn set_distribution_pip_available(distribution_source: &str) {
         // Enable if a default source is used and known to have pip installed already
         || (!distribution_source.is_empty()
             && !distribution_source.starts_with(DEFAULT_PYPY_SOURCE)
+            && env::var("PYAPP_DISTRIBUTION_PATH")
+                .unwrap_or_default()
+                .is_empty()
             && env::var("PYAPP_DISTRIBUTION_SOURCE")
                 .unwrap_or_default()
                 .is_empty())

--- a/build.rs
+++ b/build.rs
@@ -572,6 +572,11 @@ fn set_python_path(distribution_source: &str) {
     let python_path = env::var(distribution_variable).unwrap_or_default();
     let relative_path = if !python_path.is_empty() {
         python_path
+    } else if !env::var("PYAPP_DISTRIBUTION_PATH")
+        .unwrap_or_default()
+        .is_empty()
+    {
+        panic!("\n\nThe following option must be set when embedding a custom distribution: {distribution_variable}\n\n");
     } else if distribution_source.starts_with(DEFAULT_CPYTHON_SOURCE) {
         if get_python_version() == "3.7" {
             if on_windows {
@@ -678,9 +683,6 @@ fn set_distribution_pip_available(distribution_source: &str) {
         // Enable if a default source is used and known to have pip installed already
         || (!distribution_source.is_empty()
             && !distribution_source.starts_with(DEFAULT_PYPY_SOURCE)
-            && env::var("PYAPP_DISTRIBUTION_PATH")
-                .unwrap_or_default()
-                .is_empty()
             && env::var("PYAPP_DISTRIBUTION_SOURCE")
                 .unwrap_or_default()
                 .is_empty())

--- a/docs/config.md
+++ b/docs/config.md
@@ -110,7 +110,9 @@ You may indicate whether pip is already installed by setting the `PYAPP_DISTRIBU
 
 ### Embedding ### {: #distribution-embedding }
 
-You may set the `PYAPP_DISTRIBUTION_EMBED` option to `true` or `1` to embed the distribution in the executable at build time to avoid fetching it at runtime. You can set the `PYAPP_DISTRIBUTION_PATH` option to use a local path rather than fetching the source, which implicitly enables embedding. The local archive file should from [python-build-standalone](https://github.com/indygreg/python-build-standalone). Don't forget to update [`PYAPP_DISTRIBUTION_PYTHON_PATH`](#python-location) as needed.
+You may set the `PYAPP_DISTRIBUTION_EMBED` option to `true` or `1` to embed the distribution in the executable at build time to avoid fetching it at runtime.
+
+You can set the `PYAPP_DISTRIBUTION_PATH` option to use a local path rather than fetching the source, which implicitly enables embedding. The local archive should be similar to the [default distributions](#python-distribution) in that they should be a Python interpreter ready for use.
 
 ## pip
 

--- a/docs/config.md
+++ b/docs/config.md
@@ -60,7 +60,7 @@ Setting the `PYAPP_PYTHON_VERSION` option will determine the distribution used a
 | `3.10` |
 | `3.11` |
 
-The source of distributions is the [python-build-standalone](https://github.com/indygreg/python-build-standalone) project.
+The source for pre-built distributions is the [python-build-standalone](https://github.com/indygreg/python-build-standalone) project.
 
 Some distributions have [variants](https://gregoryszorc.com/docs/python-build-standalone/main/running.html) that may be configured with the `PYAPP_DISTRIBUTION_VARIANT` option:
 
@@ -110,7 +110,7 @@ You may indicate whether pip is already installed by setting the `PYAPP_DISTRIBU
 
 ### Embedding ### {: #distribution-embedding }
 
-You may set the `PYAPP_DISTRIBUTION_EMBED` option to `true` or `1` to embed the distribution in the executable at build time to avoid fetching it at runtime. You can set the `PYAPP_DISTRIBUTION_PATH` option to use a local path rather than fetching the source, which implicitly enables embedding.
+You may set the `PYAPP_DISTRIBUTION_EMBED` option to `true` or `1` to embed the distribution in the executable at build time to avoid fetching it at runtime. You can set the `PYAPP_DISTRIBUTION_PATH` option to use a local path rather than fetching the source, which implicitly enables embedding. The local archive file should from [python-build-standalone](https://github.com/indygreg/python-build-standalone). Don't forget to update [`PYAPP_DISTRIBUTION_PYTHON_PATH`](#python-location) as needed.
 
 ## pip
 

--- a/docs/config.md
+++ b/docs/config.md
@@ -112,7 +112,7 @@ You may indicate whether pip is already installed by setting the `PYAPP_DISTRIBU
 
 You may set the `PYAPP_DISTRIBUTION_EMBED` option to `true` or `1` to embed the distribution in the executable at build time to avoid fetching it at runtime.
 
-You can set the `PYAPP_DISTRIBUTION_PATH` option to use a local path rather than fetching the source, which implicitly enables embedding. The local archive should be similar to the [default distributions](#python-distribution) in that they should be a Python interpreter ready for use.
+You can set the `PYAPP_DISTRIBUTION_PATH` option to use a local path rather than fetching the source, which implicitly enables embedding. The local archive should be similar to the [default distributions](#python-distribution) in that there should be a Python interpreter ready for use.
 
 ## pip
 

